onyo/lib/filters.py

```diff
@@ -109,3 +144,132 @@ def match(self,
 
         # equivalence and regex match
         return item_value == self.value or self._re_match(str(item_value), self.value)
+
+    def _match_greater_than(self,
+                            item: Item) -> bool:
+        r"""Whether ``self.value`` is > ``Item[self.key]``.
+
+        A natural sort is used (i.e. '300' > '5').
+
+        Parameters
+        ----------
+        item
+            Item to compare against.
+        """
+
+        if self.value == UNSET_VALUE:
+            # match if:
+            #   - key is present (even if it is `None` or an empty string)
+            return self.key in item
```

Comment:
I think, it would make sense to say `<unset>` is **not** greater than `<unset>`.
Hence:
```suggestion
            return self.key in item and item[self.key] != UNSET_VALUE
```

WDYT?

Reply:
I took the literal meaning of `<unset>` (in that the key does not exist). In my opinion, something is greater than nothing. The same logic follows a bit later with non-empty lists being greater than empty lists.

However, if you see `<unset>` as `<null>`, then it's confusing. IMO, that's why we need #713, to remove this ambiguity. 

If you're ok with it, I'm hoping to work on #713 soon-ish and can address this then.

Reply:
Fine with deconflating first, yes.  In that case, the point would be about `<null>` instead.
So, greater than null shouldn't match null.

Reply:
As long as we don't forget that w/o the follow-up `_match_equal` is based on a different notion of `<unset>` than `_match_greater` in this commit - fine with merging.

Reply:
I was already planning on working on `<null>` next, so I'll actually just add it to this PR.

Then this'll all be consistent. :-)

Reply:
Done. :-D